crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -888,6 +889,73 @@ fn handle_leading_function_with_decorators_comment(comment: DecoratedComment) ->
     }
 }
 
+/// Handles comments between `**` and the variable name in dict unpacking
+/// It attaches these to the appropriate value node
+///
+/// ```python
+/// {
+///     **  # comment between `**` and the variable name
+///     value
+///     ...
+/// }
+/// ```
+fn handle_dict_unpacking_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    match comment.enclosing_node() {
+        // TODO: can maybe also add AnyNodeRef::Arguments here, but tricky to test due to
+        // https://github.com/astral-sh/ruff/issues/5176
+        AnyNodeRef::ExprDict(_) => {}
+        _ => {
+            return CommentPlacement::Default(comment);
+        }
+    };
+    if let Some(following) = comment.following_node() {
+        let preceding_end = match comment.preceding_node() {
+            Some(preceding) => preceding.end(),
+            None => comment.enclosing_node().start(),
+        };
+        let mut tokens_before = SimpleTokenizer::new(
+            locator.contents(),
+            TextRange::new(preceding_end, comment.slice().start()),
+        )
+        .skip_trivia()
+        .peekable();
+
+        while let Some(it) = tokens_before.next() {
+            if it.kind == TokenKind::Star
+                && tokens_before.peek().map(|t| t.kind) == Some(TokenKind::Star)
+            {
+                return CommentPlacement::trailing(following, comment);
+            }
+        }
```

Comment:
good q. will have a think

Reply:
actually do you think it would work if we shrink the `TextRange` (passed to the tokeniser) to 
`TextRange::new(preceding_end + TextSize::new(1), comment.slice().start()),` (to avoid the last char of the preceding token) and then just check that _all_ tokens are stars?

Reply:
The `end` of a range is always exclusive. For example, the range for the identifier `abc` in the following example is 5..8 (the 8th byte is not included). 

```python
a  = abc#comment
```

That means it should probably be safe to test if all tokens are `*` (or line continuation tokens, in case these are not handled as trivia and are allowed between the `**` )? 

Note: Simply adding `1` to the range would be unsafe if the last character of the preceding token is a non-ascii character with a representation that requires more than one byte (`TextRange` stores byte and not character ranges)

Reply:
thanks

you can see the implementation i went with now (does it look ok?). i'm pretty sure the only characters we are skipping by adding 1 are `,` or `{` which are ascii